### PR TITLE
fix(dop): delete guide webhook

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
delete guide webhook

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    delete guide webhook          |
| 🇨🇳 中文    |     删除guide相关webhook         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/dop/initialize.go [hunk 1]
```diff
@@ -999,3 +1001,28 @@ func compensateIssueStateCirculation(ep *endpoints.Endpoints) error {
...
+func deleteWebhook(bdl *bundle.Bundle) error {
+	const (
+		createHookName = "guide_create"
+		deleteHookName = "guide_delete"
+	)
+
+	hookNames := []string{createHookName, deleteHookName}
+
+	for _, v := range hookNames {
+		err := bdl.DeleteWebhook(apistructs.DeleteHookRequest{
```
Comment (sfwn) [inline modules/dop/initialize.go:1014]: `deleteHookName` will never be deleted if `createHookName` deleted success but `deleteHookName` failed at first time.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dop/initialize.go [hunk 2]
```diff
@@ -122,6 +122,8 @@ func (p *provider) Initialize(ctx servicehub.Context) error {
 
 	registerWebHook(bdl.Bdl)
 
+	deleteWebhook(bdl.Bdl)
```
Comment (sfwn) [inline modules/dop/initialize.go:125]: handle error correctly or not return error.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

File: modules/dop/initialize.go (hunk 2 above)
Comment (littlejiancc) [reply modules/dop/initialize.go]: done

**Pull request merged**